wordpress-mobile/WordPress-Android
# Introduce a base class to help handling locale changes at activity level

Check #11432 for more context on this.

Newer versions of the AppCompat library no longer support locale changes [at application level](https://issuetracker.google.com/issues/141869006#comment9), so this introduces a new base class called `LocaleAwareActivity` which is used to help handle those changes at activity level.

To test:

1. Change the app language through the app settings.
2. Make sure the changes are correctly applied.
3. Navigate through some screens and make sure the new language is kept.
4. Repeat the test with API 25 or lower and make sure it still works.

PR submission checklist:
